### ISSUE TITLE
Overwrites existing class, not compatible with splattributes

### ISSUE THREAD

**Issue by lolmaus** [lifeart/ember-class-modifier#4]
# Reproduction

(`local-class` comes from [ember-css-modules](https://github.com/salsify/ember-css-modules).)

```html
<strong
  class="foo"
  local-class="bar"
  {{class "baz"}}
></strong>

<em
  {{class "baz"}}
  ...attributes
></em>
```

# Expected

```html
<strong
  class="foo _bar_ew67wi baz"
></strong>

<em
  class="baz whatever is passed from the parent template"
></em>
```

# Actual

```html
<strong
  class="baz"
></strong>

<em
  class="baz"
></em>
```

**Comment by lifeart** [comment]
@lolmaus current `class-modifier` behavour - be single source of truth for classes, css modules do build-time transformation and updating class property values. This behavour could be fixed in ast transform time (if dom node has class modifier and local-class - add generated class as modifier argument)

**Comment by lolmaus** [comment]
@lifeart well, I can live without `<div class="foo" {{class "bar"}}>`, but without support for splattributes this addon is not usable for us.

Can't the modifier manipulate the content of `classList` rather than simply overwriting it?

**Comment by lifeart** [comment]
yes, but we need to describe all possible logic paths for such behavour (owrriding, fastboot support, attributes integration). Also, we need proof that `...attributes` applied before modifier initialized (because if after - we can get anover problems)

I was thinking about additional modifier having such functionality  - `class-list` (could be integrated into current addon)

**Comment by Duder-onomy** [comment]
I just ran into this as well. Trying to use splattributes + ember-class-modifier. 

I was able to whip this version using `classList` really quickly and everything seems to work fine:

```javascript
import { setModifierManager } from '@ember/modifier';

export default setModifierManager(
  () => ({
    createModifier() {
      return { element: null };
    },

    installModifier(state, element, args) {
      state.element = element;

      this.updateClasses(state, args);
    },

    updateModifier(state, args) {
      this.updateClasses(state, args);
    },

    updateClasses(state, args) {
      state.element.classList.add(...args.positional);

      Object.keys(args.named).forEach((key) => {
        if (args.named[key]) {
          state.element.classList.add(key);
        } else {
          state.element.classList.remove(key);
        }
      });
    },

    destroyModifier(/* state */) { },
  }),
  class ClassListManager {},
);
```

I know that it is a little naive, but I was able to pass the following tests:
* All of the tests from ember-class-modifier ( except the one about sorted class names )
* respects splattributes
* respects splattributes containing inline conditionals ( `{{if this.hasClass "class-to-add"}}` )

Is there something big I am missing here?
